### ISSUE TITLE
underscore in syntax highlighting

### ISSUE THREAD

**Issue by lorenzo-di-gregorio** [texstudio-org/texstudio#1499]
The LaTeX command \lstinline{any_underscore} is breaking the syntax highlighitng on the underscore character.
This is very annoying because the highlighting stays broken for the rest of the file.

Is there any way to locally reset the syntax highlighting?

I understand that covering the proliferation of packages will be not really feasible, but just a marker or some trick to restart the parser would be enough.

**Comment by sunderme** [comment]
short test case to avoid misunderstanding ?

**Comment by sunderme** [comment]
which version of txs ? os ?

**Comment by lorenzo-di-gregorio** [comment]
Test case:

1. Control+N to open a new window
2. In there, type

\usepackage{listings}

\lstinline{any_word} text

3. You see "any" in black, "word" and "text" in blue.

TeXstudio 3.1.0 (git 3.1.0) - Using Qt Version 5.15.2, compiled with Qt 5.15.2 R
Windows 10 Enterprise - Version 10.0.19041 Build 19041

**Comment by sunderme** [comment]
the documentation says:
\lstinline[hkey=value listi]<character ><source code><same character >

```An experimental implementation has been done to support the syntax
\lstinline[hkey=value list]{source code}.```
That syntax is not supported by txs, use the first one.

**Comment by muzimuzhi** [comment]
duplicate of #1305

**Comment by sunderme** [comment]
closed like #1305
